### PR TITLE
feat: TLS support

### PR DESCRIPTION
**Pull request by FrimIdan**

### CODE REVIEW

File: backend/pkg/rest/server.go [hunk 1]
```diff
@@ -149,13 +163,21 @@ func CreateRESTServer(port int, speculator *_speculator.Speculator, dbHandler *d
 
 	server.ConfigureFlags()
 	server.ConfigureAPI()
-	server.Port = port
+	server.Port = config.Port
+
+	// We want to serve both http and https
+	if config.EnableTLS {
+		server.EnabledListeners = []string{"https", "http"}
+		server.TLSCertificate = flags.Filename(config.TLSServerCertFilePath)
```
Comment (fishkerez) [inline backend/pkg/rest/server.go:171]: maybe we want to check that the certs actually exists?

File: charts/apiclarity/templates/deployment.yaml [hunk 2]
```diff
@@ -79,8 +89,13 @@ spec:
               value: kubernetes
             - name: FUZZER_IMAGE_NAME
               value: {{ .Values.apiclarity.modules.fuzzer.docker.image }}
+            {{- if .Values.apiclarity.tls.enabled }}
+            - name: FUZZER_PLATFORM_HOST_FROM_FUZZER
+              value: http://{{ include "apiclarity.name" . }}.{{ .Release.Namespace }}:8080/api
```
Comment (fishkerez) [inline charts/apiclarity/templates/deployment.yaml:94]: if else same values, should it be https?

File: plugins/gateway/tyk/deploy/deploy.sh [hunk 3]
```diff
@@ -2,12 +2,43 @@
 TykProxyContainerName="${TYK_PROXY_CONTAINER_NAME:-tyk-gtw}"
 TykGatewayDeploymentName="${TYK_GATEWAY_DEPLOYMENT_NAME:-tyk-gtw}"
 TykGatewayDeploymentNamespace="${TYK_GATEWAY_DEPLOYMENT_NAMESPACE:-default}"
-UpstreamTelemetryHostName="${UPSTREAM_TELEMETRY_HOST_NAME:-apiclarity-apiclarity.apiclarity:9000}"
+UpstreamTelemetryHostName="${UPSTREAM_TELEMETRY_HOST_NAME:-apiclarity-apiclarity.apiclarity}"
+UpstreamTelemetryHTTPPort="${UPSTREAM_TELEMETRY_HTTP_PORT:-9000}"
+UpstreamTelemetryTLSPort="${UPSTREAM_TELEMETRY_TLS_PORT:-9443}"
```
Comment (fishkerez) [inline plugins/gateway/tyk/deploy/deploy.sh:7]: UPSTREAM_TELEMETRY_HTTP_PORT, UPSTREAM_TELEMETRY_TLS_PORT - 
Should it be part of the env variables and helm options?

Comment (FrimIdan) [reply plugins/gateway/tyk/deploy/deploy.sh:7]: in general we are not configuring the ports via helm values so no need.

File: charts/apiclarity/templates/deployment.yaml (hunk 2 above)
Comment (FrimIdan) [reply charts/apiclarity/templates/deployment.yaml:94]: it is on purpose for now, we need that the fuzzer will support it first and then we will change to to talk to the TLS endpoint

File: backend/pkg/rest/server.go (hunk 1 above)
Comment (FrimIdan) [reply backend/pkg/rest/server.go:171]: If files are empty or does not exist if should fail during server creation 
```
			httpsServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair(string(s.TLSCertificate), string(s.TLSCertificateKey))

```

**Review by fishkerez — APPROVED** [review]

File: charts/apiclarity/templates/deployment.yaml (hunk 2 above)
Comment (akpsgit) [reply charts/apiclarity/templates/deployment.yaml:94]: worth adding a todo comment

File: charts/apiclarity/templates/post-delete-job-certs-deletion.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            - /bin/bash
+            - -c
+            - "cd /tmp &&
+{{- if .Values.trafficSource.global.httpsProxy }}
+            git config --global https.proxy {{ .Values.trafficSource.global.httpsProxy }} &&
+{{- end }}
+{{- if .Values.trafficSource.global.httpProxy }}
+            git config --global http.proxy {{ .Values.trafficSource.global.httpProxy }} &&
+{{- end }}
+            git clone -b {{ .Chart.AppVersion }} https://github.com/openclarity/apiclarity.git &&
```
Comment (akpsgit) [inline charts/apiclarity/templates/post-delete-job-certs-deletion.yaml:97]: how can this work until it's in master?

File: backend/pkg/traces/server.go [hunk 5]
```diff
@@ -55,10 +65,19 @@ func CreateHTTPTracesServer(port int, traceHandleFunc HandleTraceFunc) (*HTTPTra
 
 	server.ConfigureFlags()
 	server.ConfigureAPI()
+	server.Port = config.Port
+
+	// We want to serve both http and https
+	// TODO: need to use istio to secure the http port when the wasm is sending traces
```
Comment (akpsgit) [inline backend/pkg/traces/server.go:71]: maybe this todo should be in the wasm filter config? how does this influence the traces server?

File: charts/apiclarity/templates/post-delete-job-certs-deletion.yaml (hunk 4 above)
Comment (FrimIdan) [reply charts/apiclarity/templates/post-delete-job-certs-deletion.yaml:97]: change `{{ .Chart.AppVersion }}` to the branch name (`devel-global-specs-tls-support`)

File: backend/pkg/traces/server.go (hunk 5 above)
Comment (FrimIdan) [reply backend/pkg/traces/server.go:71]: need to inject envoy sidecar on apiclarity deployment (the traces server)

**Review by akpsgit — APPROVED** [review]

**Pull request merged**